src/components/CompareProject.vue

```diff
@@ -118,6 +119,7 @@ export default defineComponent({
   },
   created() {
     this.startTime = new Date().toISOString()
+    this.logMappingStarted(this.project.projectType)
```

Comment:
not sure here and the ones below.
Just wondering if the logging should happen in between "created" or at the very end. Just in case errors happen and the event 'mappingstarted' is triggered but actually mapping did not start

Reply:
Agreed. To be sure, I moved the event logging to the end of created() with the last commit.